File: backend/app/schemas/compensation_intelligence.py
import uuid

from pydantic import BaseModel, Field

from app.schemas.insights import InsightResponse

class CompensationObservation(BaseModel):
    opportunity_id: uuid.UUID
    role_id: uuid.UUID
    title: str
    compensation_min: float | None
    compensation_max: float | None
    midpoint: float | None
    currency: str | None
    employment_type: str | None
    remote_type: str | None
    source_basis: str


class CompensationInsight(InsightResponse):
    pass


class CompensationIntelligenceResponse(BaseModel):
    generated_at: str
    workspace_id: uuid.UUID | None = None
    target_compensation_min: float | None = None
    observations: list[CompensationObservation] = Field(default_factory=list)
    insights: list[CompensationInsight] = Field(default_factory=list)
    insufficient_data: list[str] = Field(default_factory=list)

File: frontend/src/types/compensationIntelligence.ts
import type { Insight } from "./insights";

export interface CompensationObservation {
  role_id: string;
  title: string;
  compensation_min: number | null;
  compensation_max: number | null;
  midpoint: number | null;
  currency: string | null;
  employment_type: string | null;
  remote_type: string | null;
  source_basis: string;
}

export type CompensationInsight = Insight;

export interface CompensationIntelligenceResponse {
  generated_at: string;
  workspace_id: string | null;
  target_compensation_min: number | null;
  observations: CompensationObservation[];
  insights: CompensationInsight[];
  insufficient_data: string[];
}
